### ISSUE TITLE
possible issue in ompi/mca/osc/rdma/

### ISSUE THREAD

**Issue by vasilyMellanox** [open-mpi/ompi#333]
We get hang sometimes with some test from MPICH2 package - mpich2/test/mpi/rma/locknull (the test calls for MPI_Win_lock/unlock operations), 
so we dug out that sometimes there is a situation when 0 == module->share_count 
and module->lock_status == MPI_LOCK_SHARED

so it boiled down to the next "if" statement:

 in file ompi/mca/osc/rdma/osc_rdma_passive_target.c

```c
static inline void ompi_osc_rdma_unlock_self (ompi_osc_rdma_module_t *module, ompi_osc_rdma_outstanding_lock_t *lock)
{
istead of
   if  (!(MPI_LOCK_SHARED == lock->type && 0 == --module->shared_count)) {
should be
    if (!(MPI_LOCK_SHARED == lock->type && 0 != --module->shared_count)) { 
```

 Guys, what do you think ?

Thanks,
Vasily

**Comment by jsquyres** [comment]
@vasilyMellanox FYI -- use triple-single-quotes to mark off a verbatim section of a github comment; it makes it a bunch easier to read.  Thanks!

@hjelmn Can you comment on this?

**Comment by goodell** [comment]
@vasilyMellanox I think @jsquyres meant triple-backticks (see https://help.github.com/articles/github-flavored-markdown/#fenced-code-blocks)

**Comment by hjelmn** [comment]
Hah, who wrote that code anyway.....

The check should be equivalent the one at L928. A quick check shows that they are indeed not equivalent. I will fix this in #335 and open a pull request for 1.8.

Thanks for catching this.

**Comment by hjelmn** [comment]
Ok, looks like #335 doesn't have this issue :)

**Comment by hjelmn** [comment]
Fixed in 1.8 and should be fixed now in master. Closing.
